src/Microsoft.DotNet.Build.VstsBuildsApi/VstsReleaseHttpClient.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Build.VstsBuildsApi
+{
+    internal class VstsReleaseHttpClient : IVstsDefinitionHttpClient
+    {
+        private static readonly string[] s_InstanceIdentifiableField = {
+            "_links",
+            "createdBy",
+            "createdOn",
+            "deployStep",
+            "modifiedBy",
+            "modifiedOn",
+            "owner",
+            "id",
+            "lastRelease",
+            "revision",
+        };
+
+        private const string ApiVersion = "3.1-preview.3";
+
+        private JsonHttpClient _jsonClient;
+
+        public VstsReleaseHttpClient(JObject definition, string credentials)
+        {
+            Uri uri = new Uri(GetVstsReleaseApiUrl(definition));
+
+            _jsonClient = new JsonHttpClient(
+                new Uri(string.Format("{0}://{1}/", uri.Scheme, uri.Authority)),
+                credentials);
+        }
+
+        public async Task<JObject> VstsUpdateDefinitionAsync(JObject definition)
+        {
+            string requestUri = GetRequestUri(definition, "definitions");
+            return await _jsonClient.PutAsync(requestUri, definition);
+        }
+
+        public async Task<JObject> VstsCreateDefinitionAsync(JObject definition)
+        {
+            string requestUri = GetRequestUri(definition, "definitions");
+            return await _jsonClient.PostAsync(requestUri, definition);
+        }
+
+        public async Task<JObject> VstsRetrieveDefinitionByIdAsync(JObject definition)
+        {
+            string requestUri = GetRequestUri(definition, $"definitions/{definition["id"]}");
+            return await _jsonClient.GetAsync(requestUri);
+        }
+
+        public async Task<IReadOnlyList<JObject>> VstsRetrieveDefinitionsListByNameAndPathAsync(JObject definition)
```

Comment:
This doesn't actually take into account the path, so that may be a little confusing (I don't know if Path is a concept for Release definitions).  In the Builds definitions, there needs to be some refactoring, we currently retain the Path in the local json definitions, but you can't actually create or update a definition with the Path information, and we strip it out when creating / updating.  We should consider renaming this to `VstsRetrieveDefinitionsListByName` and explicitly setting the path to "\\" to prevent https://github.com/dotnet/core-eng/issues/163.